src/components/ClusterNodes/NodeDetails.tsx

```diff
@@ -0,0 +1,765 @@
+        } else {
+            return (
+                <div>
+                    <div className="subtab-grid mb-8 cn-7 fw-6 fs-13">
+                        <div>Key|Value</div>
+                        <div>Effect</div>
+                    </div>
+                    {nodeDetail.taints.map((taint) => (
+                        <div className="subtab-grid">
+                            {renderKeyValueLabel(taint.key, taint.value)}
+                            {renderWithCopy(taint['effect'])}
+                        </div>
+                    ))}
+                </div>
+            )
+        }
+    }
+
+    const renderLabelAnnotationTaint = (): JSX.Element => {
+        return (
+            <div className="en-2 bw-1 br-4 bcn-0 mt-12">
+                <ul role="tablist" className="tab-list border-bottom pr-20 pl-20 pt-12">
+                    <li
+                        className="tab-list__tab pointer"
+                        onClick={() => {
+                            setSelectedSubTabIndex(0)
+                        }}
+                    >
+                        <div className={`mb-6 fs-13${selectedSubTabIndex == 0 ? ' fw-6 cb-5' : ' fw-4'}`}>
+                            Labels ({nodeDetail.labels.length})
+                        </div>
+                        {selectedSubTabIndex == 0 && <div className="node-details__active-tab" />}
+                    </li>
+                    <li
+                        className="tab-list__tab pointer"
+                        onClick={() => {
+                            setSelectedSubTabIndex(1)
+                        }}
+                    >
+                        <div className={`mb-6 fs-13${selectedSubTabIndex == 1 ? ' fw-6 cb-5' : ' fw-4'}`}>
+                            Annotation ({nodeDetail.annotations.length})
+                        </div>
+                        {selectedSubTabIndex == 1 && <div className="node-details__active-tab" />}
+                    </li>
+                    <li
+                        className="tab-list__tab pointer"
+                        onClick={() => {
+                            setSelectedSubTabIndex(2)
+                        }}
+                    >
+                        <div className={`mb-6 fs-13${selectedSubTabIndex == 2 ? ' fw-6 cb-5' : ' fw-4'}`}>
+                            Taints ({nodeDetail.taints?.length || 0})
+                        </div>
+                        {selectedSubTabIndex == 2 && <div className="node-details__active-tab" />}
+                    </li>
+                </ul>
+                <div className=" pr-20 pl-20 pt-12 pb-12">
+                    {selectedSubTabIndex == 0 && renderLabelTab()}
+                    {selectedSubTabIndex == 1 && renderAnnotationTab()}
+                    {selectedSubTabIndex == 2 && renderTaintTab()}
+                </div>
+            </div>
+        )
+    }
+    const renderErrorOverviewCard = (): JSX.Element => {
```

Comment:
Note: we can also explicitly mention in the return type that this function will return either JSX.Element or null i.e. `JSX.Element | null`

Reply:
Done